src/lib/wrapAISDK.ts

```diff
@@ -0,0 +1,229 @@
+import { NodeTracerProvider, BatchSpanProcessor } from '@opentelemetry/sdk-trace-node';
+import { OTLPTraceExporter } from '@opentelemetry/exporter-trace-otlp-http';
+import { defaultResource, resourceFromAttributes } from '@opentelemetry/resources';
+import { ATTR_SERVICE_NAME, ATTR_SERVICE_VERSION } from '@opentelemetry/semantic-conventions';
+import { Tracer } from '@opentelemetry/api';
+
+/**
+ * Configuration for the Scorecard AI SDK wrapper
+ */
+interface ScorecardConfig {
+  /**
+   * Scorecard project ID
+   * Defaults to SCORECARD_PROJECT_ID environment variable
+   */
+  projectId?: string;
+
+  /**
+   * Metrics to track
+   * Defaults to empty array
+   */
+  metrics?: string[];
+
+  /**
+   * Scorecard API key for authentication
+   * Defaults to SCORECARD_API_KEY environment variable
+   */
+  apiKey?: string;
+
+  /**
+   * Scorecard tracing endpoint
+   * Defaults to https://tracing.scorecard.io/otel/v1/traces
+   */
+  endpoint?: string;
+
+  /**
+   * Scorecard API base URL
+   * Defaults to https://api2.scorecard.io
+   */
+  apiBaseUrl?: string;
+
+  /**
+   * Service name for telemetry
+   * Defaults to "ai-sdk-app"
+   */
+  serviceName?: string;
+
+  /**
+   * Max export batch size for the batch span processor
+   * Defaults to 1 so the traces are exported immediately
+   */
+  maxExportBatchSize?: number;
+}
+
+let tracerProvider: NodeTracerProvider | null = null;
+let tracer: Tracer | null = null;
+const SCORECARD_API_KEY = process.env['SCORECARD_API_KEY'];
+const SCORECARD_PROJECT_ID = process.env['SCORECARD_PROJECT_ID'];
+
+/**
+ * Initialize the OpenTelemetry tracer with Scorecard configuration
+ */
+function initializeTracer(config: ScorecardConfig = {}): Tracer {
+  // Return existing tracer if already initialized
+  if (tracer && tracerProvider) {
+    return tracer;
+  }
+
+  const {
+    projectId = SCORECARD_PROJECT_ID,
+    apiKey = SCORECARD_API_KEY,
+    endpoint = 'https://tracing.scorecard.io/otel/v1/traces',
+    serviceName = 'ai-sdk-app',
+    maxExportBatchSize = 1,
+  } = config;
+
+  // Create resource with service information
+  const resource = defaultResource().merge(
+    resourceFromAttributes({
+      [ATTR_SERVICE_NAME]: serviceName,
+      [ATTR_SERVICE_VERSION]: '1.0.0',
+      'scorecard.project_id': projectId,
```

Comment:
Exclude the `scorecard.projectId` key-value pair if it's undefined:

```suggestion
      ...{projectId != null ? {'scorecard.project_id': projectId} : null),
```